Sources/TuistGraph/Graph/ConditionalGraphTarget.swift

```diff
@@ -0,0 +1,49 @@
+import Foundation
+import TSCBasic
+
+public struct GraphTargetReference: Equatable, Comparable, Hashable, CustomDebugStringConvertible, CustomStringConvertible,
+    Codable
+{
+    /// Path to the directory that contains the project where the target is defined.
+    public let graphTarget: GraphTarget
+
+    public var target: Target { graphTarget.target }
+
+    /// Platforms the target is conditionally deployed to.
+    public let condition: PlatformCondition?
+
+    public init(target: GraphTarget, condition: PlatformCondition? = nil) {
+        graphTarget = target
+        self.condition = condition
+    }
+
+    public static func < (lhs: GraphTargetReference, rhs: GraphTargetReference) -> Bool {
+        lhs.graphTarget < rhs.graphTarget
+//        guard let
```

Comment:
This needs to actually account for the condition.

Reply:
Just for me to better understand, why does the sorting of this type needs to take the platform condition into account when sorting ?

Reply:
@alexanderwe this is to ensure we have stable project generation.  With this implementation if there were two `GraphTargetReference` with the same `GraphTarget` they would be indeterminately ordered.